### ISSUE TITLE
Prevent auto-updater from executing during runtime

### ISSUE THREAD

**Issue by EmulationFanatic** [Ryujinx/Ryujinx#3801]
#### Overview:
Currently it is possible to use the auto-updater functions while a game or application is actively running. This causes unexpected behavior if the update is then applied, leading to complete failure or only partial execution of the update process, leaving the user to have to manually clean up the Ryujinx folder & re-download the latest emulator package from the website.

- During runtime, the Help > Check for Updates menu option should be disabled.
- If the auto-updater check has already found an updated build and popped up the "Do you want to update Ryujinx?" dialog box, that box should be automatically terminated if emulation is then initiated or found to be active (i.e. if the user launched a game while the auto-update check was still in progress).

#### Nature of Request:  
[ Remove all that do not apply to your request. ]
- Change

### Why would this feature be useful?
To prevent the untimely deaths of Ryujinx installations and generally improve the UX.

**Comment by EmulationFanatic** [comment]
Only partially fixed by https://github.com/Ryujinx/Ryujinx/pull/3886

**Comment by gdkchan** [comment]
The proper solution would be making the updater work regardless of when its executed. Preventing it from running is just a band-aid.

**Comment by EmulationFanatic** [comment]
> The proper solution would be making the updater work regardless of when its executed. Preventing it from running is just a band-aid.

I agree, it is a bandaid. But to make it work regardless of when its executed would essentially require an entirely new updater implementation, which we kinda need anyway.

**Comment by ethanriverpage** [comment]
Agreed. While it is a rudimentary implementation, it will hopefully be replaced sometime. I hope it will cut down on people demolishing their installations!
